src/server/game/Entities/Player/Player.cpp

```diff
@@ -1602,6 +1602,9 @@ bool Player::TeleportTo(uint32 mapid, float x, float y, float z, float orientati
             // remove all dyn objects
             RemoveAllDynObjects();
 
+            // remove all areatriggers entities
+            RemoveAllAreaTriggers();
```

Comment:
Shouldn't you resummon all area triggers that are created by Apply Aura: AreaTrigger? Spells like [Windwalking](http://www.wowhead.com/spell=157411/windwalking) are applied only once, therefore the trigger is only created once, but they should be present at any time